src/components/MainPage/MainPage.js

```diff
@@ -1,24 +1,201 @@
+  type PhotoIdentifier,
+} from '@react-native-community/cameraroll'
 
-import { TouchableButton, PhotoList, } from '../index'
+import { TouchableButton, PhotoList, SelectedPhoto, } from '../index'
+import { MAX_COUNT_LIST_PHOTOS, } from '../../constants'
 import styles from './style'
 
+import { type PhotoDataType, } from '../../types'
+
 type Props = {
   toggleCamera: () => void
 }
 
-const MainPage = ({ toggleCamera, }: Props) => (
-  <View style={styles.container}>
-    <Text style={styles.headline}>Photo List</Text>
-    <PhotoList />
-    <TouchableButton onPress={toggleCamera} style={styles.makePhotoBtn}>
-      <Text style={styles.btnText}>MAKE A PHOTO</Text>
-    </TouchableButton>
-  </View>
-)
+type ChunkInfo = { chunkNumber: number, previousChunkNumber: number }
+
+type SelectedPhotoInfo = { selectedPhotoData: ?PhotoDataType, isPhotoSelected: boolean, }
+
+type InitialPhotoState = {
+  photoList: ?Array<PhotoIdentifier>,
+  nextChunksIndicator: ?string
+}
+
+const INITIAL_PHOTO_STATE = {
+  photoList: null,
+  nextChunksIndicator: null,
+}
+
+const MainPage = ({ toggleCamera, }: Props) => {
+  const [{ photoList, nextChunksIndicator, }, setPhotoList] = useState(
+    INITIAL_PHOTO_STATE
+  );
+  ({ photoList, nextChunksIndicator, }: InitialPhotoState)
+
+  const [{ selectedPhotoData, isPhotoSelected, }, setSelectedPhoto] = useState({
+    selectedPhotoData: null,
+    isPhotoSelected: false,
+  });
+  ({ selectedPhotoData, isPhotoSelected, }: SelectedPhotoInfo)
+
+  const [{ chunkNumber, previousNumber, }, changeChunkInfo] = useState({
+    chunkNumber: 0,
+    previousNumber: 0,
+  });
+  ({ chunkNumber, previousNumber, }: ChunkInfo)
+
+  const checkAndroidPermission = async (): Promise<void> => {
+    try {
+      const permission: string =
+        PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE
+      await PermissionsAndroid.request(permission)
+      Promise.resolve()
+    } catch (error) {
+      Promise.reject(error)
+    }
+  }
+
+  const getPhoto = async (): Promise<PhotoIdentifiersPage> => {
+    const startParams: GetPhotosParams = {
+      first: MAX_COUNT_LIST_PHOTOS * 5,
+      assetType: 'Photos',
+    }
+
+    if (nextChunksIndicator) {
+      startParams.after = nextChunksIndicator
+    }
+    const data: PhotoIdentifiersPage = await CameraRoll.getPhotos(startParams)
+
+    return data
+  }
+
+  const setStartData = async (): Promise<void> => {
+    checkAndroidPermission()
+    const photoData: PhotoIdentifiersPage = await getPhoto()
 
-export { MainPage, }
+    setPhotoList({
+      photoList: photoData.edges,
+      nextChunksIndicator: photoData.page_info.end_cursor || null,
+    })
+  }
+
+  useEffect(() => {
+    setStartData()
+  }, [])
+
+  const selectPhoto = (data: PhotoDataType): void => setSelectedPhoto(
+    { selectedPhotoData: data, isPhotoSelected: true, }
+  )
+
+  const closeSelectedPhoto = (): void => setSelectedPhoto(
+    { selectedPhotoData: null, isPhotoSelected: true, }
```

Comment:
`isPhotoSelected: true` why not `false`?

Reply:
Because if **isPhotoSelected** will be **true** then the closing animation will work after closing SelectedPhoto component